content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+<div align="center">
+  `stratifySurv(Surv(time, event), strata) ~ x`.
+</div>
+
+This works for fitting the model but the syntax is rather convoluted on the left-hand side of the formula. The specification of the stratification is inconsistent with the approach taken by the survival package. Additionally, without further modification, prediction fails because variables on the left-hand side of the formula are treated as response variables by parsnip and not available at prediction time.
+
+We already need to translate from formula interface to matrix interface for this engine, and we like interfaces to be as consistent as possible here in tidymodels so we decided to go with the specification of stratification through a `strata()` term on the right-hand side of the formula for proportional hazards models, regardless of the engine.
+
+```{r strata}
+library(survival)
+
+bladder_train <- bladder[-c(1:3),]
+bladder_test <- bladder[1:3,]
+
+cox_spec <- proportional_hazards(penalty = 0.123) %>%
+    set_engine("glmnet")
+
+f_fit <- fit(cox_spec,
+             Surv(stop, event) ~ rx + size + number + strata(enum),
+             data = bladder_train)
+```
+
+
+### Prediction 
+
+We have introduced several new prediction types: time to event (`time`), linear predictor (`linear_pred`), survival probability (`survival`), and hazard (`hazard`). Some of these types are time-dependent, e.g., the survival probability is the probability to survive beyond a certain time point. In keeping with the tidymodels design principle that each row of the data set you're predicting on should give you exactly one row in the dataset of predictions, `predict(type = "survival", time)` returns a nested tibble if `time` contains multiple time points.
+
+```{r predict}
+f_pred <- predict(f_fit, new_data = bladder_test,
+                  type = "survival", time = c(5, 10, 15, 20))
+
+# 3 rows for 3 observations
+f_pred 
+# 4 rows for 4 time points
+f_pred$.pred[[1]]
+```
+
+Usually, the linear predictor is the linear combination of the predictors, weighted with the coefficients, often denoted as $x^{T} \beta$. This is what is being returned by the `glmnet` engine for `proportional_hazards()` models and by the `mboost` engine for `boost_tree()` models (e.g., with mode `"censored regression"`). For proportional hazards models, the linear predictor describes _relative_ risk, i.e., it refers to a pair of observations: the observation $x$ and a reference observation with value 0. The survival package by default centers the observation with the mean to allow for an interpretation of risk relative to a "standard" observation. For consistency with other engines, prediction of type `linear_pred` from `proportional_hazards()` models with engine `survival` also do not center the predictors.
```

Comment:
```suggestion
Usually, the linear predictor is the linear combination of the predictors, weighted with the coefficients, often denoted as $x^{T} \beta$. This is what is being returned by the `glmnet` engine for `proportional_hazards()` models and by the `mboost` engine for `boost_tree()` models (e.g., with mode `"censored regression"`). For proportional hazards models, the linear predictor describes _relative_ risk, i.e., it refers to a pair of observations: the observation $x$ and a reference observation with value 0. The survival package by default centers the observation with the mean to allow for an interpretation of risk relative to a "standard" observation. For consistency with other engines, predictions of type `linear_pred` from `proportional_hazards()` models with engine `survival` also do not center the predictors.
```

Reply:
Wait, also does *not*? This sounds confusing to me; maybe take out the "also" to clarify that something different is happening?